### ISSUE TITLE
If Container(volume_mounts) exists Container(volumes) doesnt get write out in yaml

### ISSUE THREAD

**Issue by mozarik** [argoproj-labs/hera#835]
<h2>Pre-bug-report checklist</h2>

**1. This bug can be reproduced using YAML**
- [ ] Yes [👉 Please report a bug to the Argo Workflows GitHub 👈](https://github.com/argoproj/argo-workflows/issues/new/choose)
- [x] No

**2. This bug occurs when...**
- [x] running Hera code without submitting to Argo (e.g. when exporting to YAML)
- [ ] running Hera code and submitting to Argo

<h2>Bug report</h2>

**Describe the bug**
_A clear and concise description of what the bug is:_
If i define `volume_mounts[]` and `volumes[]` in `Container()` the only thing that got write out to yaml is the `volume_mounts`. 

_Error log if applicable_:
```
No Error Log
```

**To Reproduce**
Full Hera code to reproduce the bug:
```py
with Workflow(
    generate_name="test-",
    volumes=[
        SecretVolume(name="service-account-credential",secret_name="service-account-credential")
    ],
    entrypoint="test"
) as w2:
    v_tmp_pod = Volume(
        name="tmp-pod",
        size="100Mi",
        mount_path="/tmp/pod",
    )
    init_container_example = Container(
        name="git-sync",
        volume_mounts=[
            VolumeMount(name="service-account-credential", mount_path="/secrets")
        ],
        volumes=[v_tmp_pod]
    )
    with Steps(name="test") as s:
        init_container_example()

print(w2.to_yaml())
```
output:
```yaml
apiVersion: argoproj.io/v1alpha1
kind: Workflow
metadata:
  generateName: test-
  namespace: argo-staging
spec:
  entrypoint: test
  templates:
  - container:
      image: python:3.8
      volumeMounts:
      - mountPath: /secret
        name: service-account-credential
    name: git-sync
  - name: test
    steps:
    - - name: git-sync
        template: git-sync
  volumeClaimTemplates:
  - metadata:
      name: tmp-pod
    spec:
      accessModes:
      - ReadWriteOnce
      resources:
        requests:
          storage: 100Mi
  volumes:
  - name: service-account-credential
    secret:
      secretName: service-account-credential
```

**Expected behavior**
_A clear and concise description of what you expected to happen:_

It should be volume_mounts + volumes in the `volume_mounts`

```yaml
apiVersion: argoproj.io/v1alpha1
kind: Workflow
metadata:
  generateName: test-
  namespace: argo-staging
spec:
  entrypoint: test
  templates:
  - container:
      image: python:3.8
      volumeMounts:
      - mountPath: /secrets
        name: service-account-credential
      - mountPath: /tmp/pod
        name: tmp-pod
    name: git-sync
  - name: test
    steps:
    - - name: git-sync
        template: git-sync
  volumeClaimTemplates:
  - metadata:
      name: tmp-pod
    spec:
      accessModes:
      - ReadWriteOnce
      resources:
        requests:
          storage: 100Mi
  volumes:
  - name: service-account-credential
    secret:
      secretName: service-account-credential
```

**Environment**
 - Version of Hera: 5.9.0
 - Version of Argo: [e.g. 3.5.0]

**Comment by mozarik** [comment]
Current work around

```
with Workflow(
    generate_name="test-",
    volumes=[
        SecretVolume(name="service-account-credential",secret_name="service-account-credential")
    ],
    # We do this because of this bug https://github.com/argoproj-labs/hera/issues/835
    volume_claim_templates=[
        PersistentVolumeClaim(
            metadata=ObjectMeta(name="inWorkflow"),
            spec=PersistentVolumeClaimSpec(
                access_modes=["ReadWriteOnce"],
                resources=ResourceRequirements(
                    requests={"storage": Quantity(__root__="1Gi")}
                )
            )
        )
    ],
    entrypoint="test"
) as w2:
    v_tmp_pod = Volume(
        name="tmp-pod",
        size="100Mi",
        mount_path="/tmp/pod",
    )
    init_container_example = Container(
        name="git-sync",
        volume_mounts=[
            VolumeMount(name="service-account-credential", mount_path="/secrets"),
            VolumeMount(name="inWorkflow", mount_path="/tmp/pod")
        ],
    )
    with Steps(name="test") as s:
        init_container_example()

print(w2.to_yaml())
```

Got the idea to do this after looking at this https://github.com/argoproj-labs/hera/pull/529/files#

**Comment by flaviuvadan** [comment]
Thanks for reporting @mozarik I will take a look

**Comment by flaviuvadan** [comment]
Apologies for not posting but this was fixed in 5.13.1
